File: FSharpWithAkka/Transfer/Domain/Transfer.fs
namespace Bank.Transfer.Domain

type RecipientAccountEnvironment =
   | Internal = 0
   | Domestic = 1
   | International = 2

type RecipientAccountIdentificationStrategy =
   | AccountId = 0
   | SwiftBIC = 1
   | IBAN = 2
   | NationalID = 3

type TransferRecipient = {
   LastName: string
   FirstName: string
   Identification: string
   AccountEnvironment: RecipientAccountEnvironment
   IdentificationStrategy: RecipientAccountIdentificationStrategy
   RoutingNumber: string option
   Currency: string
}

File: CSharpWithLanguageExt/Transfer/Domain/Transfer.cs
namespace Bank.Transfer.Domain;

public record TransferRecipient(
   string LastName,
   string FirstName,
   string Identification,
   RecipientAccountEnvironment AccountEnvironment,
   RecipientAccountIdentificationStrategy IdentificationStrategy,
   string? RoutingNumber = null,
   string Currency = "USD"
);

public enum RecipientAccountIdentificationStrategy {
   AccountID,
   SwiftBIC,
   IBAN,
   NationalID
}

public enum RecipientAccountEnvironment {
   Internal,
   Domestic,
   International
}